packages/protocol/contracts/rrp/authorizers/Api3TokenLock.sol

```diff
@@ -0,0 +1,305 @@
+    ///      on an airnode address
+    /// @param _airnode The airnode Address
+    /// @param _requesterAddress The requester address
+    modifier isNotBlocked(address _airnode, address _requesterAddress) {
+        require(
+            !airnodeToRequesterToBlockStatus[address(0)][_requesterAddress],
+            ERROR_REQUESTER_BLOCKED
+        );
+        require(
+            !airnodeToRequesterToBlockStatus[_airnode][_requesterAddress],
+            ERROR_REQUESTER_BLOCKED
+        );
+        _;
+    }
+
+    /// @notice Called by the admin to set the minimum locking time
+    /// @param _minimumLockingTime The new minimum locking time
+    function setMinimumLockingTime(uint256 _minimumLockingTime)
+        external
+        override
+        onlyWithRank(bytes32(0), uint256(AdminRank.Admin))
+    {
+        minimumLockingTime = _minimumLockingTime;
+        emit SetMinimumLockingTime(_minimumLockingTime);
+    }
+
+    /// @notice Called by the admin to set the locking amount
+    /// @param _lockAmount The new lock amount
+    function setLockAmount(uint256 _lockAmount)
+        external
+        override
+        onlyWithRank(bytes32(0), uint256(AdminRank.Admin))
+    {
+        require(_lockAmount != 0, ERROR_ZERO_AMOUNT);
+        lockAmount = _lockAmount;
+        emit SetLockAmount(_lockAmount);
+    }
+
+    /// @notice Called by a SuperAdmin or the airnodeAdmin to set the Block status of a airnode-requester pair
+    ///         also revokes access by setting whitelistExpiration to 0.
+    ///         Whitelist access is only revoked when the specified airnode address is specified
+    ///         otherwise specifying adress(0) would only result in the requester being unable
+    ///         to lock/unlock across all airnodes. This is an irreversible action.
+    /// @param _airnode The airnode address
+    /// @param _requesterAddress requester address
+    function blockRequester(address _airnode, address _requesterAddress)
+        external
+        override
+        onlyWithRank(
+            bytes32(abi.encode(_airnode)),
+            uint256(AdminRank.SuperAdmin)
+        )
+    {
+        airnodeToRequesterToBlockStatus[_airnode][_requesterAddress] = true;
+        IApi3RequesterRrpAuthorizer(api3RequesterRrpAuthorizer)
+            .setWhitelistExpiration(
+                bytes32(abi.encode(_airnode)),
+                _requesterAddress,
+                0
+            );
+        emit BlockRequester(_airnode, _requesterAddress, msg.sender);
+    }
+
+    /// @notice Locks API3 Tokens to gain access to Airnodes.
+    /// Airnode-requester pair gets authorized as long as there is at least one token lock for given pair.
+    /// @dev The amount to be locked is determined by a memory variable set by
+    /// the owners of the contract.
+    /// @param _airnode The airnode address
+    /// @param _requesterAddress The requester address
+    function lock(address _airnode, address _requesterAddress)
```

Comment:
Question: What would happen if _airnode is address(0)? we'll be whitelisting the requester globally? Do we want that?

Reply:
I've decided to add a zero address check for lock,unlock and withdrawExcess. On second thought they shouldnt be allowed to access the 0th address